plugins/action/generate_cloud_modules.py

```diff
@@ -0,0 +1,1359 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2020 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The action plugin file for generate_cloud_modules
+"""
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+
+import argparse
+import json
+
+import pathlib
+import re
+import yaml
+import copy
+from typing import Dict, Iterable, List, DefaultDict, Union, Optional, TypeVar, Type
+from ansible.plugins.action import ActionBase
+from ansible_collections.ansible.content_builder.plugins.plugin_utils.cloud_utils.content_library_data import content_library_static_ds
+from ansible_collections.ansible.content_builder.plugins.plugin_utils.cloud_utils.utils import (
+    format_documentation,
+    indent,
+    UtilsBase,
+    jinja2_renderer,
+    get_module_added_ins,
+    get_module_from_config,
+    python_type,
+    camel_to_snake,
+    ignore_description,
+)
+from ansible_collections.ansible.content_builder.plugins.plugin_utils.cloud_utils.generator import generate_documentation
+
```

Comment:
can we split this into vmware and aws specific and move the common functionalities to different .py file and then import common functionalities to be used by vmware and aws specific file to make this more readable and maintainable

Reply:
The generate_documentaion imported here is aws specific. For vmware, the generation is straightforward and is done directly by a function in this file. I can add relevant comments before the import.